### PR TITLE
Parallel work to expand usefulness of parallel tag

### LINKED ISSUE #204: [TASK] Allow specifying various parameters for running on cluster
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
It would be nice if we did not have to edit the outer and inner xml files to run it on the cluster.

**Describe the solution you'd like**
Allow inputs to generate the various cluster parameters.

**Describe alternatives you've considered**
Continue manually editing the xml files.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [ ] 1. Is it tagged with a type: defect or task?
- [ ] 2. Is it tagged with a priority: critical, normal or minor?
- [ ] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [ ] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [ ] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [ ] 1. If the issue is a defect, is the defect fixed?
- [ ] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [ ] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [ ] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [ ] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address?
Closes #204 

##### What are the significant changes in functionality due to this change request?
Expands the parallel tag to allow more features.
Adds a way to just run HERON with an integration test
Adds different defaults for parallel inner and outer size.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/HERON/wiki/Code-Standards) for details.
- [ ] 4. Automated Tests should pass.
- [ ] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large tes.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added, the the analytic documentation must be updated/added.
- [ ] 9. If any test used as a basis for documentation examples have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: tests/integration_tests/cluster_check/gold/outer.xml [hunk 1]
```diff
@@ -0,0 +1,140 @@
+<Simulation verbosity="all">
+  <RunInfo>
+    <JobName>Opt_Runs_o</JobName>
+    <WorkingDir>Opt_Runs_o</WorkingDir>
+    <Sequence>optimize, plot</Sequence>
+    <batchSize>3</batchSize>
+    <!-- <internalParallel>True</internalParallel>
+     <mode>mpi<runQSUB></runQSUB></mode>
+     <expectedTime>72:0:0</expectedTime>
+     <clusterParameters>-P nst -j oe</clusterParameters> -->
```
Comment (dylanjm) [inline tests/integration_tests/cluster_check/gold/outer.xml]: Should we remove this commented code from the outer.xml template file now?

**Review by dylanjm — CHANGES_REQUESTED** [review]

File: tests/integration_tests/cluster_check/gold/outer.xml (hunk 1 above)
Comment (PaulTalbot-INL) [reply tests/integration_tests/cluster_check/gold/outer.xml]: that's probably a good idea.

Comment (joshua-cogliati-inl) [reply tests/integration_tests/cluster_check/gold/outer.xml]: Removed.

**Review by dylanjm — APPROVED** [review]
Changes look good. New feature is documented.

FUTURE WORK IDEA: Add use-case examples walkthroughs to the wiki, one for local, one for HPC. Also a debugging workflow.

**Pull request merged**